### DISCUSSION TITLE
Advice on how to start coding effectively

### DISCUSSION THREAD

**Discussion by AkshatG156** [TheAlgorithms/Python#8927] [Q&A]
Hello everyone, welcoming you all.
I am new here and to the coding world. Seeking your advice on how to start coding in Python. I am already finished learning the basics such as OOPs, NumPy, Pandas but now want to start coding. A little confused and seeking your help.

**Comment by mbaranr** [comment]
Hi Akshat!
It's great news that you decided to invest your time on learning a new skill such as coding. Python seems like an excellent choice of a starting programming language because of its simplicity and versatility. I see that you have covered some really interesting topics like OOP and super useful libraries like NumPy and Pandas. Regarding your question on how to start coding effectively, I think this really comes down on what you exactly meant by "effectively". If you are asking about coding environments to boost up your productivity, I would definetely suggest trying out different IDEs or text editors until you find one that fits with your workflow the most. I personally use VSCode for the most part, because of its flexibility and already available plethora of plugins, though this can change depending on the project I am working on. However, I think that choosing an environment and getting as comfortable as you can with working on it is important.

If you were referring to other skills that might result useful to achieve your goals in the field you are interested in, you would maybe need to take a look at algorithmic analysis and data structures. I just finished my first year of a computer science degreee and now I am a research intern at a university for the summer. That being said, I can confidently say that a clear understanding of how data is handled, and how it can be stored in such a vast number of ways using different data structures (i.e. trees, graphs, priority queues, etc.) with their respective API functions, is crucial to a succesful and smooth transition into the programming world. 

Because you mentioned Pandas as one of the first libraries you decided to learn, you might be interested in data science. If that is the case, I would recommend you to create a kaggle account and play around with different free to access datasets. Often, picking a project that you might find interesting and starting to work on it (even if it seems too much of a challenge at the beginning) by breaking it up into sub-problems will definetely help you nourish the right mindset required to suceed in a field like this. I would encourage you to stay ambitious and preserve the sense of curiosity that you might be feeling right now. 

You just made the first step towards a beautiful journey and I wish you the best of lucks!

**Reply by AkshatG156** [reply]
Thank you my friend. I really appreciate your advice and it clears most of my thoughts. I want to be a data scientist and it is a career transition I am making after 6.5 years of working as a business consultant. Not sure how much time I am gonna take to get a new job as I lost my job in March.
Do you think I should learn to program in Python to be a data scientist or just the stats/probability and pandas utility will suffice?

**Reply by mbaranr** [reply]
That is awesome! Regarding your question, I think it really depends on what are your goals as a data scientist. If you are looking to develop software to analyze data for other people to use, I think learning about python as a programming language is very important. If you are only going to use pandas to analyze data for your own use, maybe pandas utility will suffice.

**Reply by AkshatG156** [reply]
Thank you. As per your advice, I have started with basics and would want to learn until I start using machine learning concepts. I would also want to stay in touch with you for future advice.
